src/consumption.cpp

```diff
@@ -205,191 +217,159 @@ morale_type player::allergy_type( const item &food ) const
-        if( !saprophage && !has_trait( trait_id( "SAPROVORE" ) ) &&
-            !maybe_query( _( "This %s smells awful!  Eat it?" ) ) ) {
-            return ROTTEN;
+
+    if( food.rotten() ) {
+        const bool saprovore = has_trait( trait_id( "SAPROVORE" ) );
+        if( !saprophage && !saprovore ) {
+            consequences.emplace_back( ROTTEN, _( "This is rotten an smells awful!" ) );
         }
-    } else if( saprophage && edible && !food.has_flag( "FERTILIZER" ) &&
-               !maybe_query( _( "Really eat that %s?  Your stomach won't be happy." ) ) ) {
+    } else if( saprophage && edible && !food.has_flag( "FERTILIZER" ) ) {
         // Note: We're allowing all non-solid "food". This includes drugs
         // Hardcoding fertilizer for now - should be a separate flag later
         //~ No, we don't eat "rotten" food. We eat properly aged food, like a normal person.
         //~ Semantic difference, but greatly facilitates people being proud of their character.
-        maybe_print( m_info, _( "It's too fresh, let it age a little first." ) );
-        return ROTTEN;
+        consequences.emplace_back( ROTTEN, _( "Your stomach won't be happy (not rotten enough)." ) );
     }
 
     if( edible && has_effect( effect_nausea ) ) {
-        if( !maybe_query(
-                _( "You still feel nauseous and will probably puke it all up again.  Eat anyway?" ) ) ) {
-            return ALLERGY;
-        }
+        consequences.emplace_back( ALLERGY,
+                                   _( "You still feel nauseous and will probably puke it all up again." ) );
     }
 
-    // Print at most one of those
-    bool overfull = false;
-    if( overeating ) {
-        overfull = !maybe_query( _( "You're full.  Force yourself to eat?" ) );
-    } else if( ( ( nutr > 0 && temp_hunger < capacity ) ||
-                 ( comest->quench > 0 && temp_thirst < capacity ) ) &&
-               !food.has_infinite_charges() ) {
-        // This first section is only for flavor msg, skip if we wouldn't print anything
-        if( interactive &&
-            ( slimespawner ||
-              ( gourmand && has_active_mutation( trait_id( "GOURMAND" ) ) ) ||
-              ( eathealth && has_active_mutation( trait_id( "EATHEALTH" ) ) ) ) ) {
-            add_msg_if_player( _( "You're full, but you cram it into your mouth without a second thought." ) );
-        } else {
-            overfull = !maybe_query( _( "You will not be able to finish it all.  Consume it?" ) );
+    const int nutr = nutrition_for( food.type );
+    const int quench = comest->quench;
+    const int temp_hunger = get_hunger() - nutr;
+    const int temp_thirst = get_thirst() - quench;
+
+    if( has_active_mutation( trait_id( "HIBERNATE" ) ) ) {
+        if( ( get_hunger() >= -60 && get_thirst() >= -60 ) &&
+            ( temp_hunger < -60 || temp_thirst < -60 ) ) {
+            consequences.emplace_back( TOO_FULL, _( "You're adequately fueled. Prepare for hibernation?" ) );
```

Comment:
What will be the context of this message?
You removed questions from other messages, why not this one?